File: plugins/spigot/TitleManager/src/main/java/io/puharesource/mc/titlemanager/backend/hooks/PluginHook.java
package io.puharesource.mc.titlemanager.backend.hooks;

import org.bukkit.Bukkit;
import org.bukkit.plugin.Plugin;

public abstract class PluginHook {
    private String pluginName;

    public PluginHook(String pluginName) {
        this.pluginName = pluginName;
    }

    public boolean isEnabled() {
        return Bukkit.getPluginManager().isPluginEnabled(pluginName);
    }

    public Plugin getPlugin() {
        return Bukkit.getPluginManager().getPlugin(pluginName);
    }
}

File: src/main/kotlin/io/puharesource/mc/titlemanager/internal/functionality/placeholder/PluginHook.kt
package io.puharesource.mc.titlemanager.internal.functionality.placeholder

import io.puharesource.mc.titlemanager.internal.debug
import org.bukkit.Bukkit
import org.bukkit.plugin.Plugin

abstract class PluginHook constructor(private val pluginName: String) {
    init {
        if (!isEnabled()) {
            debug("$pluginName is not enabled, disabling placeholders related to the plugin.")
        }
    }

    fun isEnabled() : Boolean = Bukkit.getPluginManager().isPluginEnabled(pluginName)

    fun getPlugin() : Plugin = Bukkit.getPluginManager().getPlugin(pluginName)!!
}